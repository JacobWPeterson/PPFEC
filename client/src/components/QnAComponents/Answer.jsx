import React from 'react';
import Photos from './Photos.jsx'


class Answer extends React.Component {
  constructor(props) {
    super(props)
    this.state ={

    }
  }


  render () {
    return(
      <div>
          A: {this.props.answer.body}
          <div>{this.props.answer.photos.length !=  0 ? <Photos photos={this.props.answer.photos}/>:  null}</div>
      </div>
    )
  }

}
export default Answer;