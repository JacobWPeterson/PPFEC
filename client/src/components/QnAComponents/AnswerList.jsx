import React from 'react';
import Answer from './Answer.jsx';

const AnswerList = (props) => (
  <div>

    {props.answers.map((answer) => (<Answer answer={answer} key={answer.id} id={answer.id} />))}

  </div>
);

export default AnswerList;
