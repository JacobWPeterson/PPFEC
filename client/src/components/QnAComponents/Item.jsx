import React from 'react';
import styled from 'styled-components';
import AnswerList from './AnswerList.jsx';
import MoreA from './MoreA.jsx';
import Modal from './Modal.jsx';
import FormA from './FormAnswer.jsx';

const TopWrapper = styled.div`
  display: flex;
  flex-direction: row;
  font-size: 20px;
  padding: 2vh 0;
  justify-content: space-between;
  align-items: center;
`;

const BottomWrapper = styled.div`
  display: flex;
  flex-direction: row;
`;

const Helper = styled.div`
  display: flex;
  font-size: 15px;
  min-width: 245px;
  padding-left: 4vw;
  color: #424242;
`;

const Yes = styled.div`
  padding: 0 2px 0 1vw;
  text-decoration: underline;
  &:hover {
    cursor: pointer;
    color: #80CCC4;
    transform: scale(1.1);
  };
`;

const AddAButton = styled.div`
  padding-left: 1vw;
  text-decoration: underline;
  &:hover {
    cursor: pointer;
    color: #80CCC4;
    transform: scale(1.1);
  };
`;

const Helpfulness = styled.div`
  padding-right: 1vw;
`;

const A = styled.div`
  font-size: 20px;
  padding-right: 1vw;
  font-weight: 700;
  color: #424242;
`;

const Q = styled.div`
  font-weight: 700;
  padding-right: 1vw;
  color: #424242;
`;

const QB = styled.h4`
  display: flex;
  flex-direction: row;
  font-weight: 700;
  font-size: 1.3rem;
`;

const ListContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

class Item extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      answers: [],
      dd: [],
      max: 0,
      displayed: 0,
      buttonDisplay: true,
      modal: false,
    };
    this.loadTwoItems = this.loadTwoItems.bind(this);
    this.handleClick = this.handleClick.bind(this);
    this.exitModal = this.exitModal.bind(this);
  }

  componentDidMount() {
    Object.entries(this.props.item.answers).forEach((answer) => {
      this.state.answers.push(answer[1]);
    });
    this.setState((prevState) => ({
      answers: prevState.answers,
    }));
    this.loadTwoItems();
  }

  handleClick() {
    this.setState({ modal: true });
  }

  exitModal() {
    this.setState({ modal: false });
  }

  loadTwoItems() {
    this.state.max += 2;
    while (this.state.displayed < this.state.max) {
      if (!this.state.answers[this.state.displayed]) {
        break;
      }
      this.state.dd.push(this.state.answers[this.state.displayed]);
      this.state.displayed += 1;
    }
    if (!this.state.answers[this.state.displayed]) {
      this.state.buttonDisplay = false;
    }
    this.setState((prevState) => ({
      dd: prevState.dd,
      displayed: prevState.displayed,
      max: prevState.max,
      buttonDisplay: prevState.buttonDisplay,
    }));
  }

  render() {
    return (
      <div>
        <TopWrapper>
          <QB>
            <Q>Q:</Q>
            {this.props.item.question_body}
          </QB>
          <Helper>
            <div>Helpful?</div>
            <Yes
              onClick={this.handleClick}
            >
              Yes
            </Yes>
            {this.state.modal ? (
              <Modal content={(
                <FormA
                  func={this.exitModal}
                />
              )}
              />
            ) : null}
            <Helpfulness>{`(${this.props.item.question_helpfulness})`}</Helpfulness>
            <div>|</div>
            <AddAButton
              onClick={this.handleClick}
            >
              Add Answer
            </AddAButton>
          </Helper>
        </TopWrapper>
        <BottomWrapper>
          <A>A:</A>
          <ListContainer>
            <AnswerList answers={this.state.dd} />
            <MoreA func={this.loadTwoItems} display={this.state.buttonDisplay} />
          </ListContainer>
        </BottomWrapper>
      </div>
    );
  }
}

export default Item;
