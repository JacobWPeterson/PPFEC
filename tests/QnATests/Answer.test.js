import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Item from '../../client/src/components/QnAComponents/Item.jsx';
import AnswerList from '../../client/src/components/QnAComponents/AnswerList.jsx';
import Answer from '../../client/src/components/QnAComponents/Answer.jsx';

describe('One Answer Test', () => {
  const props = {
    answer: {
      answerer_name: 'sadf',
      body: 'asdf',
      date: '2021-03-08T00:00:00.000Z',
      helpfulness: 0,
      id: 1443618,
      photos: ['https://pngimg.com/uploads/mario/mario_PNG53.png', 'https://pngimg.com/uploads/mario/mario_PNG53.png'],
    },
  };

  const answerComp = mount(<Answer {...props} debug />);

  it('The form should contain Helpful component', () => {
    expect(answerComp.find(<div>Helpfulish</div>)).toBeTruthy();
  });

  it('The form should not contain Helpfulish component', () => {
    expect(answerComp.find(<div>Helpfulish</div>)).toEqual({});
  })
});
