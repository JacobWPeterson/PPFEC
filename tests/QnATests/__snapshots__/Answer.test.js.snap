// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`One Answer Test Should render an Answer correctly 1`] = `
<<<<<<< HEAD
<Answer
  answer={
    Object {
      "answerer_name": "sadf",
      "body": "asdf",
      "date": "2021-03-08T00:00:00.000Z",
      "helpfulness": 0,
      "id": 1443618,
      "photos": Array [
        "https://pngimg.com/uploads/mario/mario_PNG53.png",
        "https://pngimg.com/uploads/mario/mario_PNG53.png",
      ],
    }
  }
  debug={true}
>
  <div>
    <Answer__Body>
      <div
        className="Answer__Body-sc-3mvxjy-6 gnmbP"
      >
        asdf
      </div>
    </Answer__Body>
    <div>
      <Photos
        photos={
          Array [
            "https://pngimg.com/uploads/mario/mario_PNG53.png",
            "https://pngimg.com/uploads/mario/mario_PNG53.png",
          ]
        }
      >
        <div>
          <Photos__Photo
            alt="https://pngimg.com/uploads/mario/mario_PNG53.png"
            key="https://pngimg.com/uploads/mario/mario_PNG53.png0"
            src="https://pngimg.com/uploads/mario/mario_PNG53.png"
          >
            <img
              alt="https://pngimg.com/uploads/mario/mario_PNG53.png"
              className="Photos__Photo-sc-1homjbu-0 dZzZNL"
              src="https://pngimg.com/uploads/mario/mario_PNG53.png"
            />
          </Photos__Photo>
          <Photos__Photo
            alt="https://pngimg.com/uploads/mario/mario_PNG53.png"
            key="https://pngimg.com/uploads/mario/mario_PNG53.png1"
            src="https://pngimg.com/uploads/mario/mario_PNG53.png"
          >
            <img
              alt="https://pngimg.com/uploads/mario/mario_PNG53.png"
              className="Photos__Photo-sc-1homjbu-0 dZzZNL"
              src="https://pngimg.com/uploads/mario/mario_PNG53.png"
            />
          </Photos__Photo>
        </div>
      </Photos>
    </div>
    <Answer__BottomContainer>
      <div
        className="Answer__BottomContainer-sc-3mvxjy-0 kfNGYX"
      >
        <Answer__NameDate>
          <div
            className="Answer__NameDate-sc-3mvxjy-1 jcOBlI"
          >
            By sadf on  March, 08 2021
          </div>
        </Answer__NameDate>
        |
        <Answer__Helpful>
          <div
            className="Answer__Helpful-sc-3mvxjy-2 iSIJxX"
          >
            Helpful?
          </div>
        </Answer__Helpful>
        <Answer__Yes
          onClick={[Function]}
        >
          <div
            className="Answer__Yes-sc-3mvxjy-3 bpatiV"
            onClick={[Function]}
          >
            Yes
          </div>
        </Answer__Yes>
        <Answer__Help>
          <div
            className="Answer__Help-sc-3mvxjy-4 gOKkij"
          >
            (0)
          </div>
        </Answer__Help>
        |
        <Answer__Report
          onClick={[Function]}
        >
          <div
            className="Answer__Report-sc-3mvxjy-5 fkJrli"
            onClick={[Function]}
          >
            Report
          </div>
        </Answer__Report>
      </div>
    </Answer__BottomContainer>
  </div>
</Answer>
`;

exports[`One Answer Test Should render correctly 1`] = `
=======
>>>>>>> 02ee0ec6c214bf9bfe9c4a4496ac203921d04516
<Answer
  answer={
    Object {
      "answerer_name": "sadf",
      "body": "asdf",
      "date": "2021-03-08T00:00:00.000Z",
      "helpfulness": 0,
      "id": 1443618,
      "photos": Array [
        "https://pngimg.com/uploads/mario/mario_PNG53.png",
        "https://pngimg.com/uploads/mario/mario_PNG53.png",
      ],
    }
  }
  debug={true}
>
  <div>
    <Answer__Body>
      <div
        className="Answer__Body-sc-3mvxjy-6 gnmbP"
      >
        asdf
      </div>
    </Answer__Body>
    <div>
      <Photos
        photos={
          Array [
            "https://pngimg.com/uploads/mario/mario_PNG53.png",
            "https://pngimg.com/uploads/mario/mario_PNG53.png",
          ]
        }
      >
        <div>
          <Photos__Photo
            alt="https://pngimg.com/uploads/mario/mario_PNG53.png"
            key="https://pngimg.com/uploads/mario/mario_PNG53.png0"
            src="https://pngimg.com/uploads/mario/mario_PNG53.png"
          >
            <img
              alt="https://pngimg.com/uploads/mario/mario_PNG53.png"
              className="Photos__Photo-sc-1homjbu-0 dZzZNL"
              src="https://pngimg.com/uploads/mario/mario_PNG53.png"
            />
          </Photos__Photo>
          <Photos__Photo
            alt="https://pngimg.com/uploads/mario/mario_PNG53.png"
            key="https://pngimg.com/uploads/mario/mario_PNG53.png1"
            src="https://pngimg.com/uploads/mario/mario_PNG53.png"
          >
            <img
              alt="https://pngimg.com/uploads/mario/mario_PNG53.png"
              className="Photos__Photo-sc-1homjbu-0 dZzZNL"
              src="https://pngimg.com/uploads/mario/mario_PNG53.png"
            />
          </Photos__Photo>
        </div>
      </Photos>
    </div>
    <Answer__BottomContainer>
      <div
        className="Answer__BottomContainer-sc-3mvxjy-0 kfNGYX"
      >
        <Answer__NameDate>
          <div
            className="Answer__NameDate-sc-3mvxjy-1 jcOBlI"
          >
            By sadf on  March, 08 2021
          </div>
        </Answer__NameDate>
        |
        <Answer__Helpful>
          <div
            className="Answer__Helpful-sc-3mvxjy-2 iSIJxX"
          >
            Helpful?
          </div>
        </Answer__Helpful>
        <Answer__Yes
          onClick={[Function]}
        >
          <div
            className="Answer__Yes-sc-3mvxjy-3 bpatiV"
            onClick={[Function]}
          >
            Yes
          </div>
        </Answer__Yes>
        <Answer__Help>
          <div
            className="Answer__Help-sc-3mvxjy-4 gOKkij"
          >
            (0)
          </div>
        </Answer__Help>
        |
        <Answer__Report
          onClick={[Function]}
        >
          <div
            className="Answer__Report-sc-3mvxjy-5 fkJrli"
            onClick={[Function]}
          >
            Report
          </div>
        </Answer__Report>
      </div>
    </Answer__BottomContainer>
  </div>
</Answer>
`;
